shm/alcohol.cpp

```diff
@@ -0,0 +1,17 @@
+#include "alcohol.hpp"
+
+Alcohol::Alcohol(std::string name, size_t amount, size_t alcoholByVolume)
+    : Cargo(name, amount, alcoholByVolume_){};
```

Comment:
What happens here? You pass the uninitialized variable `alcoholByVolume_`.
What's more `Cargo`  class should not know anything about this parameter